openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/BuildConfigOperationsImpl.java

```diff
@@ -235,7 +235,10 @@ public void writeTo(BufferedSink sink) throws IOException {
         }
       };
 
-      OkHttpClient newClient = client.newBuilder().readTimeout(timeout, timeoutUnit).build();
+      OkHttpClient newClient = client.newBuilder()
+        .readTimeout(timeout, timeoutUnit)
+        .writeTimeout(timeout, timeoutUnit)
```

Comment:
Struggling to find a clean way to implement this. How do we want the the BuildConfigs DSL to look?

At the moment you'd do something like:

```
buildConfigs()
  .withName("foo")
  .instantiateBinary()
  .withTimeout(60, TimeUnit.SECONDS)
```

Any ideas on the best way to introduce the writeTimeout value?

Simplest way would probably be to extend withTimeout (for this builder) to include a writeTimeout arg, so you're always forced to specify both a read & write timeout value.